feat: Post pending screens

cmaddox5 opened the pull request:
**Asana task**: [[Perm config] Post pending screens to S3](https://app.asana.com/0/1185117109217413/1205973195471505/f)

This PR connects the server to the client when posting pending screens. Screens configured in the client are added/updated/deleted from the pending config after clicking "Review Screens". This work depends on `local_pending.json` existing in your local screens repo. You can also verify S3 logic by switching `pending_screens_config_fetcher: Screenplay.PendingScreensConfig.Fetch.Local` to `pending_screens_config_fetcher: Screenplay.PendingScreensConfig.Fetch.S3` in `dev.exs`.

hannahpurcell commented on test/test_helper.exs:
```diff
@@ -1 +1,4 @@
+Mox.defmock(Screenplay.RoutePatterns.Mock, for: Screenplay.RoutePatterns.Behaviour)
```
I'm excited about this mock management library!!

cmaddox5 replied:
I think code will be much cleaner with it because we won't need any of the dependency injection that Screens uses. Much less parameters with default values. 🎉

cmaddox5 commented on lib/screenplay/config/config.ex:
```diff
@@ -1,37 +1,62 @@
-        error
-    end
+  defp gl_eink_config_reducer(place_and_screens, acc) do
+    {place_id, %{"updated_screens" => updated_screens, "new_screens" => new_screens}} =
```
I agree, I need to take some time and comment this file a bit. I will make sure to do that. As for what `updated_screens` is referring to, that list will hold all of the pending configs that existed previously and were "updated" before publishing. Live screens will never be part of either of the lists on this line. So if you click [Add Screen](https://www.figma.com/file/FKWRVrSdoMjZOLorhuykvc/Permanent-Configuration?type=design&node-id=2417-24475&mode=design&t=PZjvHVbu7Turkw6R-4) in the workflow and fill out a new screen, that will show up in `new_screens`. If you enter that page while there are already pending screens and you make updates to them, those will show up in `updated_screens`.

cmaddox5 commented on lib/screenplay/config/config.ex:
```diff
@@ -1,37 +1,62 @@
-        config
-        |> Map.delete(screen_id)
-        |> S3Fetch.put_screens_config()
+  defp get_config_reducer(:gl_eink_v2), do: &gl_eink_config_reducer/2
```
😎

cmaddox5 commented:
@hannahpurcell I added a few comments and changed some variable names to help code readability. Let me know if you need comments added anywhere specific.
